pkg/asset/agent/image/agentimage.go

```diff
@@ -131,6 +126,44 @@ func (a *AgentImage) updateIgnitionImg(ignition []byte) error {
 	return nil
 }
 
+// updateIgnitionInfo updates igninfo.json with the current size of
+// the ignition buffer.
+func (a *AgentImage) updateIgnitionInfo(ignitionSize int) error {
+	ignInfoPath := filepath.Join(a.tmpPath, "coreos", "igninfo.json")
+	ignInfoRaw, err := os.ReadFile(ignInfoPath)
+	if err != nil {
+		return err
+	}
+
+	type ignInfo struct {
+		File   string `json:"file"`
+		Length int64  `json:"length"`
+		Offset int64  `json:"offset"`
+	}
+	var info ignInfo
+	err = json.Unmarshal([]byte(ignInfoRaw), &info)
+	if err != nil {
+		return err
+	}
+
+	info.Length = int64(ignitionSize)
+	newIgnInfoRaw, err := json.Marshal(info)
+	if err != nil {
+		return err
+	}
+
+	err = os.WriteFile(ignInfoPath, newIgnInfoRaw, 0)
+	if err != nil {
+		return err
+	}
+
+	if ignitionSize > ignitionMaxSize {
```

Comment:
Shouldn't the file only be updated as above if this is true?

Reply:
The idea was to set it always regardless of the size, given that such piece of info is used by `coreos-installer`, so to have a consistent ouput. The log trace was required to inform the user that the final ISO is going to grow

Reply:
Shouldn't we compare against the actual size, rather than hardcoding it?

Reply:
If you refer to `ignitionMaxSize` probably it's not a good name, it should be something more like `ignitionEmbedAreaSize`, I'll change it

Reply:
Yeah, but also I meant that CoreOS might in theory change the size, so it would be better to read it out of the ISO.

Reply:
Probably a remote case, but yeah, better stay on the safe side